storage/core/src/main/java/com/hortonworks/streamline/storage/StorageManager.java

```diff
@@ -68,6 +68,14 @@
     void addOrUpdate(Storable storable) throws StorageException;
 
     /**
+     * Updates an existing {@link Storable}. Throws {@link StorageException} if the
+     * storable could not be updated.
+     *
+     * @param storable the storable
+     */
+    void update(Storable storable);
```

Comment:
It updates existing rows. It will not attempt to add a row automatically. I am not sure why we have `addOrUpdate`, it has confusing semantics and the caller cant know whether the row existed and got updated or just got added. We should have had only `add` and `update` since we have separate POST and PUT calls for adding and updating.